### PR TITLE
Added scribe as a logging system, addresses #137

### PR DESCRIPTION
**Pull request by sgpthomas**
This seems like a nice simple logging framework. It advertises seamless support with Scala.js. It is pretty simply to add logging for new types of objects. I've done this for Positionals so that you can log/warn fuse source locations.

### CODE REVIEW

File: src/main/scala/Main.scala [hunk 1]
```diff
@@ -53,10 +59,15 @@ object Main {
 
   def main(args: Array[String]): Unit = {
 
+
     type ErrString = String
 
     parser.parse(args, Config(null)) match {
       case Some(conf) => {
+        val myFormatter: Formatter = formatter"[$levelColored] $message$newLine"
```
Comment (rachitnigam) [inline src/main/scala/Main.scala]: Move the logger setup code to either (1) start of main, or (2) a separate function in Main.scala.

File: src/main/scala/Syntax.scala [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
 object Syntax {
 
   import Errors._
 
+  /** Makes all positionals logable by scribe */
+  implicit object PositionalLoggable extends Loggable[(String,Positional)] {
+    override def apply(value: (String, Positional)) = {
+      new output.TextOutput(s"${value._1}\n${value._2.pos.longString}")
+    }
+  }
```
Comment (rachitnigam) [inline src/main/scala/Syntax.scala]: How is this used? Do I simply do `logger.warn(expr)`? Can you write an example (in the github thread) of how you use it?

**Comment by sgpthomas** [comment]
It's simply `scribe.warn("warning")`. Similarly for, `debug`, `info`, `error`. You also don't have to print strings, so for example I added code to print `(String, Positional)`, and so `scribe.warn(("some message", expr))` will work.

**Comment by rachitnigam** [comment]
Is it possible to simply use the `withPos` function defined for Errors instead of the implicit? In general, I shy away from having global implicit (which this seems to be) and implicit classes on common things like pairs.

**Comment by sgpthomas** [comment]
Yeah it's definitely possible to do without the implicit class. You can just use something like
`scribe.warn(Errors.withPos("hi there", Some(expr.pos)));`. Maybe I'm misunderstanding something but this implicit class isn't defining a new method on tuples, it's extending the `Loggable` trait defined in `scribe`. Basically, it looks like the `scribe.info` and related functions take in a `Loggable[M]` value. The `Loggable` trait then has information  about how to transform `M` into something loggable. So in this case the implicit class is providing a new way to print out `(String, Positional)` values. I'm not seeing why this is so bad, though I understand why it's bad in general to add a lot of implicit functionality.

**Comment by sgpthomas** [comment]
This is the library code for making strings loggable
```scala
  implicit object StringLoggable extends Loggable[String] {
    def apply(value: String): LogOutput = new TextOutput(value)
  }
```

**Comment by rachitnigam** [comment]
Oh, I see. Seems like it's a [Scala typeclass](https://blog.scalac.io/2017/04/19/typeclasses-in-scala.html). The methods here:

https://github.com/outr/scribe/blob/544886b43624acd7801461abbad2701e1828d644/core/shared/src/main/scala/scribe/LoggerSupport.scala#L17

try to view the parameters with a specific typeclass. I'm fine with having that logic in, however, I don't think it belongs in the Syntax.scala file. Make a Logger.scala for the logging configuration and the implicit objects and we can use it to store logging specific stuff.

**Comment by rachitnigam** [comment]
LGTM!

**Pull request merged**